order-backend/src/main/java/io/eventuate/examples/tram/ordersandcustomers/orders/OrderFactory.java

```diff
@@ -4,13 +4,23 @@
 import io.eventuate.tram.events.subscriber.DomainEventDispatcher;
 import io.eventuate.tram.events.subscriber.DomainEventDispatcherFactory;
 import io.micronaut.context.annotation.Factory;
+import org.hibernate.SessionFactory;
+import org.springframework.orm.hibernate5.HibernateTransactionManager;
+import org.springframework.orm.jpa.JpaTransactionManager;
+import org.springframework.transaction.support.TransactionTemplate;
 
 import javax.inject.Singleton;
+import javax.persistence.EntityManagerFactory;
 
 @Factory
 public class OrderFactory {
   @Singleton
   public DomainEventDispatcher domainEventDispatcher(CustomerEventConsumer customerEventConsumer, DomainEventDispatcherFactory domainEventDispatcherFactory) {
     return domainEventDispatcherFactory.make("consumerServiceEvents", customerEventConsumer.domainEventHandlers());
   }
+
+  @Singleton
```

Comment:
Remind me why HibernateTransactionManager is used here?
What transaction manager does @Transactional use?

Reply:
Chris, HibernateTransactionManager used by @Transactional. Micronaut uses hibernate. So that transactional manager is required to integrate together all transactions. I checked that it works.

Reply:
So does `@Transactional` use the TemplateTemplate (and the `HibernateTransactionManager`) that you are creating here? Or is a HibernateTransactionManager automatically created elsewhere? If it is then can't you use that instead of creating a second one.

Reply:
Chris, SqlTableBasedDuplicateMessageDetector uses TransactionTemplate to insert messages in transaction. By default TransactionTemplate is created with DataSourceTransactionManager. But example app, uses hibernate transactions. And to join to that transaction via template I created TransactionTemplate with HibernateTransactionManager.

Reply:
Sure. But why are you creating HibernateTransactionManager that is only used by the  TransactionTemplate .
You should use the one that exists already (and is used by @Transactional)

Reply:
And perhaps TransactionTemplate should be created with whatever TransactionalManager already exists

Reply:
yes, it works